File: backend/app/schemas/subscription.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.subscriptions.plans import PlanCode, SubscriptionStatus


class WorkspacePlanLimitsRead(BaseModel):
    model_config = ConfigDict(extra="forbid")

    members: int
    projects: int
    ai_requests_per_month: int


class WorkspacePlanUsageRead(BaseModel):
    model_config = ConfigDict(extra="forbid")

    members: int
    projects: int
    ai_requests_this_month: int


class WorkspaceSubscriptionRead(BaseModel):
    model_config = ConfigDict(extra="forbid")

    plan: PlanCode
    status: SubscriptionStatus
    current_period_start: datetime | None
    current_period_end: datetime | None
    cancel_at_period_end: bool
    limits: WorkspacePlanLimitsRead
    usage: WorkspacePlanUsageRead

File: frontend/src/types/subscription.ts
export type PlanCode = "free" | "pro";

export type SubscriptionStatus =
  "active" | "inactive" | "canceled" | "past_due" | "trialing" | "incomplete";

export type WorkspacePlanLimits = {
  members: number;
  projects: number;
  ai_requests_per_month: number;
};

export type WorkspacePlanUsage = {
  members: number;
  projects: number;
  ai_requests_this_month: number;
};

export type WorkspaceSubscription = {
  plan: PlanCode;
  status: SubscriptionStatus;
  current_period_start: string | null;
  current_period_end: string | null;
  cancel_at_period_end: boolean;
  limits: WorkspacePlanLimits;
  usage: WorkspacePlanUsage;
};
